zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/DataStoreManager.scala

```diff
@@ -133,47 +138,38 @@ class DataStoreManager(metaDataset: String,
     val dataSetRawSchema = registerTable.schema
 
     if(!metaData.contains(dataSetName)){
-
       try{
-        val primaryKey = dataSetRawSchema.primaryKey.map(dataSetRawSchema.getField(_).get)
-        val timeField = dataSetRawSchema.getField(dataSetRawSchema.timeField) match {
-          case Some(f) =>
-            if(f.isInstanceOf[TimeField]){
-              f.asInstanceOf[TimeField]
-            } else {
-              throw new QueryParsingException("Time field of " + dataSetRawSchema.typeName + "is not in TimeField format.\n")
+        dataSetRawSchema.toResolved match {
+          case temporal: TemporalSchema =>
+            collectStats(dataSetName, temporal) onComplete {
+              case Success((interval, size)) =>
+                val currentDateTime = new DateTime()
+                val stats = Stats(currentDateTime, currentDateTime, currentDateTime, size)
+                val registerDataSetInfo = DataSetInfo(dataSetName, None, temporal, interval, stats)
+
+                metaData.put(dataSetName, registerDataSetInfo)
+                flushMetaData()
+                sender ! DataManagerResponse(true, "Register Finished: temporal dataset " + dataSetName + " has successfully registered.\n")
+
+              case Failure(f) =>
+                throw CollectStatsException(f.getMessage)
             }
-          case None =>
-            throw new QueryParsingException("Time field is not specified for " + dataSetRawSchema.typeName + ".\n")
-        }
-        val resolvedSchema = Schema(dataSetRawSchema.typeName, dataSetRawSchema.dimension, dataSetRawSchema.measurement, primaryKey, timeField)
-
-        collectStats(dataSetName, resolvedSchema) onComplete {
-          case Success((interval, size)) =>
-            val currentDateTime = new DateTime()
-            val stats = Stats(currentDateTime, currentDateTime, currentDateTime, size)
-            val registerDataSetInfo = DataSetInfo(dataSetName, None, resolvedSchema, interval, stats)
-
-            metaData.put(dataSetName, registerDataSetInfo)
-            flushMetaData()
-            sender ! DataManagerResponse(true, "Register Finished: dataset " + dataSetName + " has successfully registered.\n")
 
-          case Failure(_) =>
-            //FIXME Need to introduce static dataset to handle this
+          case static: StaticSchema =>
             val currentDateTime = new DateTime()
-            val fakeStats = Stats(currentDateTime, currentDateTime, currentDateTime, 1000000)
-            val fakeStartDate = new DateTime(2005, 3, 26, 12, 0, 0, 0)
+            val fakeStats = Stats(currentDateTime, currentDateTime, currentDateTime, 1000)
+            val fakeStartDate = new DateTime(2017, 1, 1, 12, 0, 0, 0)
```

Comment:
What is the purpose of this fakeTimestamp? Could we ignore the stat info for Static datasets?